src/main/scala/com.snowplowanalytics.sauna/responders/hipchat/SendRoomNotificationResponder.scala

```diff
@@ -0,0 +1,238 @@
+   *                      its' `streamContent`.
+   * @return A [[RoomNotificationReceived]] if the command is valid and needs to be
+   *         processed by this responder, None if command is invalid or should be skipped.
+   */
+  override def extractEvent(observerEvent: ObserverBatchEvent): Option[RoomNotificationReceived] = {
+    observerEvent.streamContent match {
+      case Some(is) =>
+        val commandJson = Json.parse(Source.fromInputStream(is).mkString)
+        extractCommand[RoomNotification](commandJson) match {
+          case Right((envelope, data)) =>
+            processEnvelope(envelope) match {
+              case None =>
+                Some(RoomNotificationReceived(data, observerEvent))
+              case Some(error) =>
+                logger ! Notification(error)
+                None
+            }
+          case Left(error) =>
+            logger ! Notification(error)
+            None
+        }
+      case None =>
+        logger ! Notification("No stream present, cannot parse command")
+        None
+    }
+  }
+
+  /**
+   * Send a valid room notification using the API wrapper.
+   *
+   * @param event The event containing a room notification.
+   */
+  override def process(event: RoomNotificationReceived): Unit =
+    hipchat.sendRoomNotification(event.roomNotification).onComplete {
+      case Success(message) => context.parent ! RoomNotificationSent(event, s"Successfully sent HipChat notification: $message")
+      case Failure(error) => logger ! Notification(s"Error while sending HipChat notification: $error")
+    }
+}
+
+object SendRoomNotificationResponder {
+  /**
+   * A responder event denoting that a HipChat room notification was received by
+   * the responder.
+   *
+   * @param roomNotification A room notification represented as a case class.
+   * @param source           The observer event that triggered this responder event.
+   */
+  case class RoomNotificationReceived(
+    roomNotification: RoomNotification,
+    source: ObserverBatchEvent) extends ResponderEvent[ObserverBatchEvent]
+
+  /**
+   * A responder result denoting that a HipChat room notification was successfully sent
+   * by the responder.
+   *
+   * @param source  The responder event that triggered this.
+   * @param message A success message.
+   */
+  case class RoomNotificationSent(
+    source: RoomNotificationReceived,
+    message: String) extends ResponderResult
+
+  /**
+   * A self-describing JSON structure, containing a reference to a JSON
+   * schema and data that should validate against the given schema.
+   *
+   * @param schema A self-describing schema key.
+   * @param data   Data that should comply to the schema.
+   */
+  case class SelfDescribing(
+    schema: SchemaKey,
+    data: JsValue)
+
+  implicit val selfDescribingReads: Reads[SelfDescribing] = (
+    (JsPath \ "schema").read[String].map[SchemaKey](schemaKey => SchemaKey.fromUri(schemaKey).get) and
```

Comment:
`Option.get` is always bad. If somehow invalid schema key will sneak - user will get cryptic traceback about `None.get`. 

I'd use something like:

```scala
implicit val foo: Reads[SchemaKey] = Reads {
  case JsString(s) => SchemaKey.fromUri(s) match {
    case Some(key) => JsSuccess(key)
    case None => JsError() // Something about invalid SchemaKey
  }
  case _ => JsError()
}
```